### DISCUSSION TITLE
Not updated docs ? : Creating a ModelSerializer without either the 'fields' attribute or the 'exclude' attribute has been deprecated since 3.3.0

### DISCUSSION THREAD

**Discussion by Guilouf** [encode/django-rest-framework#8242] [Potential Issue] [answered]
But in the docs, there is this paragraph https://github.com/encode/django-rest-framework/blob/master/docs/api-guide/serializers.md#specifying-fields-explicitly, where obviously there is no `fields` or `exclude`:
```python
class AccountSerializer(serializers.ModelSerializer):
    url = serializers.CharField(source='get_absolute_url', read_only=True)
    groups = serializers.PrimaryKeyRelatedField(many=True)

    class Meta:
        model = Account
```

When i try something similar, i get the assertion error `Creating a ModelSerializer without either the 'fields' attribute or the 'exclude' attribute has been deprecated since 3.3.0, and is now disallowed. Add an explicit fields = '__all__' to the CalendarTaskSerializer serializer.`.

Is it an issue with docs not updated ? How to customize a model serializer, with for example renamed fields ?

**Comment by lovelydinosaur** [comment, marked as answer]
Ah good spot, thanks. Looks like it needs updating, yup. You're welcome to open a pull request adding `fields = ...` there, or else I'll update it. 👍

**Reply by Guilouf** [reply]
Ok, i will transfer to an issue and then a PR !

**Reply by lovelydinosaur** [reply]
No need to issue it - welcome to just go straight ahead with the PR. :)

**Reply by Guilouf** [reply]
Ok, but i might have another issue, i tried to test the code with a simple model:

```python
# models.py
class Task(models.Model):
    name = models.CharField()

   @property
    def get_prop(self):
        return 'eyeyey'

# serializer.py
class TaskSerializer(serializers.ModelSerializer):
    title = serializers.CharField(source='get_prop', read_only=True),

    class Meta:
        model = Task
        fields = '__all__'
```

I get the name in the result page, but not `title`. I tried with specifying fields, and with `read_only_fields`, but no luck. What am i missing there ?

**Reply by Guilouf** [reply]
Ok, i found, i got to list the `get_prop` property in `fields`. It does not work with `__all__` by the way

**Reply by Guilouf** [reply]
There: https://github.com/encode/django-rest-framework/pull/8243
